### PR TITLE
Core - Add outgoing spectator transmissions

### PR DESCRIPTION
**Pull request by Soldia1138**
**When merged this pull request will:**
- Add the ability to hear outgoing transmissions on radios selected as Specator Radios

**Things to do and comments**
- Spectator Radios are persistent after quitting spectator -> This leads to double radio sounds if the own radio is in Spectator Radios
- As outgoing transmissions have radio effects (they are exactly like if the spectator has a radio) it's hard to distinguish which radio is currently transmitting. We should maybe add a visualization on that.

### CODE REVIEW

File: addons/sys_core/fnc_speaking.sqf [hunk 1]
```diff
@@ -100,7 +100,7 @@ if !(GVAR(keyedMicRadios) isEqualTo []) then {
     private _compiledParams = HASH_CREATE;
     {
         private _recRadio = _x;
-        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)}) then {
+        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)} || toLower(_recRadio) in ACRE_SPECTATOR_RADIOS) then {
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_speaking.sqf]:
```suggestion
        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)} || {toLower(_recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```

File: addons/sys_core/fnc_speaking.sqf [hunk 2]
```diff
@@ -128,7 +128,7 @@ if !(GVAR(keyedMicRadios) isEqualTo []) then {
                 // if (!GVAR(speaking_cache_valid)) then {
                 private _sourceRadios = _sources select _forEachIndex;
                 private _hearableRadios = [_recRadio, "handleMultipleTransmissions", _sourceRadios] call EFUNC(sys_data,transEvent);
-                if (GVAR(fullDuplex)) then {
+                if (GVAR(fullDuplex) || toLower(_recRadio) in ACRE_SPECTATOR_RADIOS) then {
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_speaking.sqf]:
```suggestion
                if (GVAR(fullDuplex) || {toLower(_recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```

File: addons/sys_core/script_component.hpp [hunk 3]
```diff
@@ -3,7 +3,7 @@
 #include "\idi\acre\addons\main\script_mod.hpp"
 
 // #define USE_DEBUG_EXTENSIONS
-// #define DEBUG_MODE_FULL
+//#define DEBUG_MODE_FULL
```
Comment (TheMagnetar) [inline addons/sys_core/script_component.hpp]:
```suggestion
// #define DEBUG_MODE_FULL
```

**Review by TheMagnetar — CHANGES_REQUESTED** [review]
Didn't we change all radio IDs  lower case?

File: addons/sys_core/fnc_speaking.sqf (hunk 1 above)
Comment (Soldia1138) [reply addons/sys_core/fnc_speaking.sqf]: At least that broke my tests yesterday. So there are still leftovers where we have mixed cases

Comment (Soldia1138) [reply addons/sys_core/fnc_speaking.sqf]: Scratch that

**Comment by Soldia1138** [comment]
> Didn't we change all radio IDs lower case?

At least that broke my tests yesterday. So there are still leftovers where we have mixed cases

File: addons/sys_core/fnc_speaking.sqf [hunk 4]
```diff
@@ -100,7 +100,7 @@ if !(GVAR(keyedMicRadios) isEqualTo []) then {
     private _compiledParams = HASH_CREATE;
     {
         private _recRadio = _x;
-        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)}) then {
+        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)} || {toLower(_recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```
Comment (jonpas) [inline addons/sys_core/fnc_speaking.sqf]:
```suggestion
        if (_recRadio != ACRE_BROADCASTING_RADIOID || {GVAR(fullDuplex)} || {(toLower _recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```

File: addons/sys_core/fnc_speaking.sqf [hunk 5]
```diff
@@ -128,7 +128,7 @@ if !(GVAR(keyedMicRadios) isEqualTo []) then {
                 // if (!GVAR(speaking_cache_valid)) then {
                 private _sourceRadios = _sources select _forEachIndex;
                 private _hearableRadios = [_recRadio, "handleMultipleTransmissions", _sourceRadios] call EFUNC(sys_data,transEvent);
-                if (GVAR(fullDuplex)) then {
+                if (GVAR(fullDuplex) || {toLower(_recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```
Comment (jonpas) [inline addons/sys_core/fnc_speaking.sqf]:
```suggestion
                if (GVAR(fullDuplex) || {(toLower _recRadio) in ACRE_SPECTATOR_RADIOS}) then {
```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by TheMagnetar — APPROVED** [review]

**Comment by jonpas** [comment]
Pushing back until this gets tested more.

**Review by jonpas — APPROVED** [review]

**Comment by jonpas** [comment]
Will be tested as part of dev-build.

**Pull request merged**